components/pagination/pagination.ts

```diff
@@ -0,0 +1,166 @@
+import { Component, Input, Output, OnChanges, SimpleChanges, EventEmitter, HostBinding } from "@angular/core";
+
+@Component({
+    selector: "sui-pagination",
+    template: `
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(1)" [class.disabled]="page===1">
+    <span><i class="angle double left icon"></i></span>
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page-1)" [class.disabled]="!hasPrevious()">
+    <span><i class="angle left icon"></i></span>
+</a>
+<a *ngFor="let p of pages" class="item" [class.active]="p===page" (click)="setPage(p)">
+    {{ p }}
+</a>
+<a *ngIf="hasNavigation" class="item" (click)="setPage(page+1)" [class.disabled]="!hasNext()">
+    <span><i class="angle right icon"></i></span>
+</a>
+<a *ngIf="hasBoundaryLinks" class="item"  (click)="setPage(pageCount)" [class.disabled]="page===pageCount">
+    <span><i class="angle double right icon"></i></span>
+</a>
+`
+})
+export class SuiPagination implements OnChanges {
+
+    @HostBinding("class.ui")
+    @HostBinding("class.pagination")
+    @HostBinding("class.menu")
+    private _paginationClasses:boolean;
+
+    // Public members
+    public pageCount:number;
+
+    @Output()
+    public pageChange:EventEmitter<number>;
+
+    // Private members
+    private _maxSize?:number;
```

Comment:
I've just added strict null check support to the library, and it introduced lots of places where `| null` was used and lots of places where `| undefined` was used. I noticed that all of the library functions use `| undefined`, so I've opted to move the entire codebase towards using `undefined` as the only 'nothing' keyword. Therefore, mark this is done and ignore my above comment 😄